Fluid/ParserContext.cs

```diff
@@ -37,5 +37,12 @@ public void ExitBlock()
         {
             CurrentBlock = _blocks.Pop();
         }
+        /// <summary>
+        /// Invoked at the end of parsing to ensure we closed all blocks
+        /// </summary>
+        public bool AreInBlock()
+        {
+            return(_blocks.Count > 0);
```

Comment:
One last thing, please inline this code directly in DefaultFluidParser. And remove the changes in this file.

Reply:
_blocks would need to be public or internal to be accessed from DefaultFluidParser. Looking at CurrentBlock may work, but I'm not sure if that would cover every case and it would need to be two check I think.
Being that this is probably not going to be called in a tight loop, although I could be wrong if used in something like orchard, I don't think the performance difference will be much. I think it's more a question of code readability and maintainability.

Simplest option I can think of is make _blocks internal instead of private.